### PR TITLE
[style] include missing header

### PR DESCRIPTION
**Pull request by bukepo**
always include openthread-br/config.h so that OTBR_CONFIG_FILE will always take effect.

### CODE REVIEW

File: src/agent/application.hpp [hunk 1]
```diff
@@ -34,6 +34,8 @@
 #ifndef OTBR_AGENT_APPLICATION_HPP_
 #define OTBR_AGENT_APPLICATION_HPP_
 
+#include "openthread-br/config.h"
```
Comment (superwhd) [inline src/agent/application.hpp:37]: The config.h header is empty as of now. What's the use case of it?

Comment (superwhd) [reply src/agent/application.hpp:37]: What about removing the macro `OTBR_CONFIG_FILE`? We may not have to keep it unless there's anybody using it.

Comment (bukepo) [reply src/agent/application.hpp:37]: actually it's used on Android platform.

**Review by superwhd — APPROVED** [review]
LGTM 👍

File: .github/workflows/macOS.yml [hunk 2]
```diff
@@ -63,4 +63,4 @@ jobs:
         brew reinstall boost cmake cpputest dbus jsoncpp ninja protobuf pkg-config
     - name: Build
       run: |
-        OTBR_OPTIONS='-DOTBR_BORDER_AGENT=OFF -DOTBR_MDNS=OFF -DOT_FIREWALL=OFF -DOTBR_DBUS=OFF' ./script/test build
+        OTBR_OPTIONS='-DOTBR_BORDER_AGENT=OFF -DOTBR_MDNS=OFF -DOT_FIREWALL=OFF -DOTBR_DBUS=OFF -DOTBR_FEATURE_FLAGS=OFF' ./script/test build
```
Comment (bukepo) [inline .github/workflows/macOS.yml:66]: disable on macOS until it's fixed. @morningboata

**Review by wgtdkp — APPROVED** [review]
LGTM :+1:

File: .github/workflows/macOS.yml [hunk 3]
```diff
@@ -63,4 +63,4 @@ jobs:
         brew reinstall boost cmake cpputest dbus jsoncpp ninja protobuf pkg-config
     - name: Build
       run: |
-        OTBR_OPTIONS='-DOTBR_BORDER_AGENT=OFF -DOTBR_MDNS=OFF -DOT_FIREWALL=OFF -DOTBR_DBUS=OFF' ./script/test build
+        OTBR_OPTIONS='-DOTBR_BORDER_AGENT=OFF -DOTBR_MDNS=OFF -DOT_FIREWALL=OFF -DOTBR_DBUS=OFF -DOTBR_FEATURE_FLAGS=OFF -DOTBR_TELEMETRY_DATA_API=OFF' ./script/test build
```
Comment (bukepo) [inline .github/workflows/macOS.yml:66]:
```suggestion
        OTBR_OPTIONS='-DOTBR_BORDER_AGENT=OFF -DOTBR_MDNS=OFF -DOT_FIREWALL=OFF -DOTBR_DBUS=OFF' ./script/test build
```

Comment (bukepo) [inline .github/workflows/macOS.yml:66]:
```suggestion
        OTBR_OPTIONS='-DOTBR_BORDER_AGENT=OFF -DOTBR_MDNS=OFF -DOT_FIREWALL=OFF -DOTBR_DBUS=OFF' ./script/test build
```

**Review by jwhui — APPROVED** [review]

**Pull request merged**